lib/semantics/resolve-names.cpp

```diff
@@ -1612,6 +1613,12 @@ void DeclTypeSpecVisitor::Post(
 void DeclTypeSpecVisitor::MakeNumericType(TypeCategory category, int kind) {
   SetDeclTypeSpec(context().MakeNumericType(category, kind));
 }
+ 
+void DeclTypeSpecVisitor::CheckForAbstractType(const Symbol &typeSymbol) {
```

Comment:
If this function is only called once, I don't think it provides much value over its inline textual expansion.

Reply:
Agreed.  It's called twice, once in the `Post` function for `parser::TypeSpec` and again in the `Post` function for `parser::DeclarationTypeSpec`.  It just felt wrong to copy the code, so I created a separate function.